hack/make/build-rpm

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+DEST=$1
+
+# subshell so that we can export PATH without breaking other things
+(
+	source "$(dirname "$BASH_SOURCE")/.integration-daemon-start"
+
+	# we need to wrap up everything in between integration-daemon-start and
+	# integration-daemon-stop to make sure we kill the daemon and don't hang,
+	# even and especially on test failures
+	didFail=
+	if ! {
+		set -e
+
+		# TODO consider using frozen images for the dockercore/builder-rpm tags
+		
+		rpmVersion="${VERSION//-/'~'}"
+		# if we have a "-dev" suffix or have change in Git, let's make this package version more complex so it works better
+		if [[ "$VERSION" == *-dev ]] || [ -n "$(git status --porcelain)" ]; then
+			gitUnix="$(git log -1 --pretty='%at')"
+			gitDate="$(date --date "@$gitUnix" +'%Y%m%d.%H%M%S')"
+			gitCommit="$(git log -1 --pretty='%h')"
+			gitVersion="git${gitDate}.0.${gitCommit}"
+			# gitVersion is now something like 'git20150128.112847.0.17e840a'
+			rpmVersion="$rpmVersion~$gitVersion"
+		fi
+		
+		# if go-md2man is available, pre-generate the man pages
+		./docs/man/md2man-all.sh -q || true
+		# TODO decide if it's worth getting go-md2man in _each_ builder environment to avoid this
+
+		# TODO add a configurable knob for _which_ rpms to build so we don't have to modify the file or build all of them every time we need to test
+		for dir in contrib/builder/rpm/*/; do
+			version="$(basename "$dir")"
+			suite="${version##*-}"
+
+			image="dockercore/builder-rpm:$version"
+			if ! docker inspect "$image" &> /dev/null; then
+				( set -x && docker build -t "$image" "$dir" )
+			fi
+
+			mkdir -p "$DEST/$version"
+			cat > "$DEST/$version/Dockerfile.build" <<-EOF
+				FROM $image
+				COPY . /usr/src/docker-core
+				RUN mkdir -p /root/rpmbuild/SOURCES
+				WORKDIR /root/rpmbuild
+				RUN ln -sfv /usr/src/docker-core/hack/make/.build-rpm SPECS
+				RUN tar -cvz -C /usr/src -f docker-core.tar.gz docker-core \
+					&& mv docker-core.tar.gz /root/rpmbuild/SOURCES
+				WORKDIR /root/rpmbuild/SPECS
+				RUN rpmbuild -ba --define '_rpmversion $rpmVersion' --define '_version $VERSION' docker-core.spec
+			EOF
+			cp -a "$DEST/$version/Dockerfile.build" . # can't use $DEST because it's in .dockerignore...
```

Comment:
Ooh, if we fixed #12494, building with `-f some/Dockerfile` that's in `.dockerignore` wouldn't be an issue anymore.  Gonna make a comment over there pointing this out too. :+1: